src/client_factory.rs

```diff
@@ -113,17 +113,19 @@ impl ClientFactory {
             "Creating reader group {:?} to read data from stream {:?}",
             reader_group_name, stream
         );
+        let scope = stream.scope.clone();
         let rg_config = ReaderGroupConfigBuilder::default().add_stream(stream).build();
-        ReaderGroup::create(reader_group_name, rg_config, self.clone()).await
+        ReaderGroup::create(scope, reader_group_name, rg_config, self.clone()).await
```

Comment:
This logic is wrong. A reader group can read from many streams and they can even be in different scopes. So the scope of the reader group should NOT be the same as the scope for the stream.